packages/@lwc/integration-not-karma/test/component/lifecycle-callbacks/index.spec.js

```diff
@@ -319,11 +323,18 @@ describe('connectedCallback/renderedCallback timing when reconnected', () => {
                 resetTimingBuffer();
 
                 document.body.appendChild(elm);
-                expect(window.timingBuffer).toEqual(
-                    !lwcRuntimeFlags.DISABLE_NATIVE_CUSTOM_ELEMENT_LIFECYCLE
-                        ? ['parent:connectedCallback', 'child:connectedCallback']
-                        : ['parent:connectedCallback']
-                );
+
+                const expected = ['parent:connectedCallback'];
+
+                if (lwcRuntimeFlags.DISABLE_DETACHED_REHYDRATION) {
+                    expected.push('parent:renderedCallback');
+                }
+
+                if (!lwcRuntimeFlags.DISABLE_NATIVE_CUSTOM_ELEMENT_LIFECYCLE) {
+                    expected.push('child:connectedCallback');
+                }
```

Comment:
This changed because previously, both the parent and child were being rehydrated while they were both disconnected from the DOM.

When the elements are re-inserted into the DOM, the VMs are not longer marked as dirty so rehydration is skipped.

The `renderedCallback` did not fire previously because the parent element was not connected to the DOM.